src/Shared/TryParseMethodCache.cs

```diff
@@ -121,146 +122,200 @@ public bool HasTryParseMethod(ParameterInfo parameter)
 
         private static MethodInfo GetEnumTryParseMethod(bool preferNonGenericEnumParseOverload)
         {
-            var staticEnumMethods = typeof(Enum).GetMethods(BindingFlags.Public | BindingFlags.Static);
+            MethodInfo? methodInfo = null;
 
-            // With NativeAOT, if there's no static usage of Enum.TryParse<T>, it will be removed
-            // we fallback to the non-generic version if that is the case
-            MethodInfo? genericCandidate = null;
-            MethodInfo? nonGenericCandidate = null;
-
-            foreach (var method in staticEnumMethods)
+            if (preferNonGenericEnumParseOverload)
             {
-                if (method.Name != nameof(Enum.TryParse) || method.ReturnType != typeof(bool))
+                methodInfo = typeof(Enum).GetMethod(
+                                nameof(Enum.TryParse),
+                                BindingFlags.Public | BindingFlags.Static,
+                                new[] { typeof(Type), typeof(string), typeof(object).MakeByRefType() });
+            }
+            else
+            {
+                foreach (var method in typeof(Enum).GetMethods(BindingFlags.Public | BindingFlags.Static))
```

Comment:
```csharp
methodInfo = typeof(Enum).GetMethod(nameof(Enum.TryParse), 1, new Type[] { typeof(String), Type.MakeGenericMethodParameter(0).MakeByRefType() });
```

Might be a bit shorter. Also, trimming will know to keep just the specific method.

`typeof(Enum).GetMethods(BindingFlags.Public | BindingFlags.Static)` keeps all public static methods on `System.Enum`.

Reply:
I wasn't sure how to write this.  Thanks!

Reply:
I feel like I've just leveled up my reflection powers...

Reply:
> I feel like I've just leveled up my reflection powers...

Glad I helped! But also, I'm a bit scared.

Reply:
🤣

Reply:
@pranavkm come look at this magic.